### DISCUSSION TITLE
Compatibility with Tencent Cloud COS

### DISCUSSION THREAD

**Discussion by YKDZ** [outline/outline#6413] [Self Hosting]
I am trying out S3 storage and have created storage buckets using the Tencent Cloud COS service. For all file upload requests, I am getting the following 403 error:
```
<Error>
	<Code>AccessDenied</Code>
	<Message>Condition key x-cos-algorithm doesn&apos;t match the value AWS4-HMAC-SHA256</Message>
	<ServerTime>2024-01-22T09:30:36Z</ServerTime>
	<Resource>/public/xxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxx.jpg</Resource>
	<RequestId>xxx=</RequestId>
	<TraceId>xxx</TraceId>
</Error>
```
Does this mean that Outline does not support Tencent Cloud COS services?
As a comparison, I also use Alibaba Cloud OOS service and everything works fine.
The environment variables I use are as follows:
```
# AWS credentials (optional in development)
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_REGION=ap-xxx
AWS_S3_UPLOAD_BUCKET_URL=https://docs-xxx-xxx-1301767611.cos.ap-xxx.myqcloud.com
AWS_S3_ACCELERATE_URL=https://docs-xxx-xxx-1301767611.cos.ap-xxx.myqcloud.com
AWS_S3_UPLOAD_BUCKET_NAME=docs-xxx-xxx-1301767611
AWS_S3_UPLOAD_MAX_SIZE=134217728
AWS_S3_FORCE_PATH_STYLE=false
# uploaded s3 objects permission level, default is private
# set to "public-read" to allow public access
AWS_S3_ACL=private
```

**Comment by tommoor** [comment]
Known compatibility is [documented here](https://docs.getoutline.com/s/hosting/doc/file-storage-N4M0T6Ypu7#h-s3-compatible-services). Tencent COS not being included means it hasn't been confirmed to work.
